controlplane/pkg/model/model.go

```diff
@@ -21,14 +22,45 @@ type Dataplane struct {
 	LocalMechanisms  []*local.Mechanism
 	RemoteMechanisms []*remote.Mechanism
 }
+type ClientConnectionState int8
+const (
+	ClientConnection_Ready = 0
+	ClientConnection_Requesting = 1
+	ClientConnection_Healing	= 2
+	ClientConnection_Closing	= 3
+	ClientConnection_Closed	= 4
+)
 
 type ClientConnection struct {
 	ConnectionId string
 	Xcon         *crossconnect.CrossConnect
 	RemoteNsm    *registry.NetworkServiceManager
 	Endpoint     *registry.NSERegistration
 	Dataplane    *Dataplane
-	IsClosing    bool
+	ConnectionState ClientConnectionState
+	Request      nsm.NSMRequest
+}
+
+func (cc *ClientConnection) GetId() string {
+	if cc == nil {
+		return ""
+	}
+	return cc.ConnectionId
+}
+
+func (cc *ClientConnection) GetNetworkService() string {
+	if cc == nil {
+		return ""
+	}
+	return cc.Endpoint.GetNetworkService().GetName()
+}
+
+func (cc *ClientConnection) GetSourceConnection() nsm.NSMConnection {
```

Comment:
Should the working be swapped here: `GetConnectionSource`?

Reply:
Yes, thanks.